### DISCUSSION TITLE
How to get external ip address

### DISCUSSION THREAD

**Discussion by ArthurZhou** [vlang/v#18165] [Questions and Answers]
How to get external ip address

**Comment by jlmess77** [comment]
You can use IP-API.com to get it by x.json2 module. Check the documents at https://ip-api.com/docs/api:json and https://modules.vlang.io/x.json2.html

**Reply by Airr** [reply]
Another approach:  https://github.com/vlang/v/discussions/15426#discussion-4304361

**Reply by Wajinn** [reply]
The code needs to be updated to use `!` otherwise use `or {}` instead of `?` .

**Reply by jlmess77** [reply]
I was a little bored and learning at the same time. Here's what I wrote:
```
module main

import x.json2
import net.http

struct GetIp {
	query string
}

fn main() {
	url := http.get('http://ip-api.com/json/') or {
		println('Failed to fetch JSON from ip-api.com.')
		return
	}

	ip_response := json2.decode[GetIp](url.body) or {
		println('Failed to decode JSON response.')
		return
	}

	println(ip_response.query)
}
```
